test/metabase/models/user_parameter_value_test.clj

```diff
@@ -6,44 +6,47 @@
    [metabase.test :as mt]
    [toucan2.core :as t2]))
 
-(deftest user-parameter-value-batch-upsert-test
-  (mt/with-temp [:model/Dashboard {dashboard-id :id} {}]
-    (let [value!   (fn [parameters]
-                     (upv/batched-upsert! (mt/user->id :rasta) dashboard-id parameters))
-          value-fn (fn []
-                     (t2/select-fn->fn :parameter_id :value
-                                       :model/UserParameterValue
-                                       :user_id (mt/user->id :rasta) :dashboard_id dashboard-id))]
+(set! *warn-on-reflection* true)
 
-      (testing "insert upv if value is non-nil"
-        (value! [{:id "param1" :value 1}
-                 {:id "param2" :value "string"}
-                 {:id "param3" :value ["A" "B" "C"]}])
-        (is (= {"param1" 1
-                "param2" "string"
-                "param3" ["A" "B" "C"]}
-               (value-fn))))
+(deftest user-parameter-value-store-test
+  (mt/test-helpers-set-global-values!
+    (mt/with-temp [:model/Dashboard {dashboard-id :id} {}]
+      (let [store!      (fn [parameters]
+                          (upv/store! (mt/user->id :rasta) dashboard-id parameters)
+                          (Thread/sleep 110))
```

Comment:
you can set `synchronous-batch-updates` to true so the grouper consumes every single entry it takes right away.